### ISSUE TITLE
Home connect started using a new platform called 'Single key'

### ISSUE THREAD

**Issue by socialhugo** [ekutner/home-connect-hass#123]
I reveived a e-mail from siemens/home connect that they will tranform the platform to 'single key'
https://singlekey-id.com/nl/home/

Since this time the regular login via api call doesn't work anymore.
Maybe the applications has to check with another platform for login creditials?

I did create a single key login etc etc. but it didn't work.
Conclusion: because of this the integration doesn't work anymore.

Anyone else has this problem?

**Comment by ekutner** [comment]
They integration is using OAuth for authenticating with Home Connect, it shouldn't matter how you are logging in. If you can login to the Home Connect developer website and configured everything as the instructions say it should work.

**Comment by socialhugo** [comment]
Thanks for the reply and all your work
I am getting this message:
" unauthorized_client " and "request rejected by client authorization authority (developer portal)"

Ever since they start changing it to 'single key' in home connect.
I tried to install everything from the beginning but no luck

**Comment by sroettering** [comment]
I am getting the same error `request rejected by client authorization authority (developer portal)` when I setup this integration. The core integration stopped working for me so I tried to switch to this one but no luck here. I did not switch to the SingleKey ID if that is important to know.

**Comment by ekutner** [comment]
Please double check that everything is configured as documented in the README. If it is then post your HC app config, the integration config and the URL where you get the error.

**Comment by socialhugo** [comment]
[Homeconnect.pdf](https://github.com/ekutner/home-connect-hass/files/9893473/Homeconnect.pdf)

I made a pdf with the configuration see file attached

**Comment by softhuntermike** [comment]
Could anyone update if the problem is fixed and how to fix? I still can't get my dishwasher connected to HA after the "single key" was rolled out. Thanks.

**Comment by ekutner** [comment]
@socialhugo I'm not sure why it's not working, the whole OAuth handshake is something that HA is handling by itself.  Just to verify, is the account configured there for testing (socialhugo) the same account you use when logging in to the HC mobile app?
In your PDF the next page after adding the integration in HA is the error message. Where you asked to login and and grant permissions before getting the error?

PS, once you manage to login you should delete this app and register a new one because it's secret key is now public.

**Comment by socialhugo** [comment]
Is the account configured there for testing (socialhugo) the same account you use when logging in to the HC mobile app?
Answer: yes. 

In your PDF the next page after adding the integration in HA is the error message. Where you asked to login and and grant permissions before getting the error?
Answer: yes, i have to login with my e-mail adress and credentials. Which works, but I haven't put it in the document.

PS, once you manage to login you should delete this app and register a new one because it's secret key is now public.
Answer: I already deleted it and made a new one.

I do not think it's the fault of your integration. Home connect has changed something because of the new 'single key' authentication. I will contact them also. 

Greetings and thanks for the effort

**Comment by ekutner** [comment]
I just tried it and it's working fine for me.
In the "Home Connect user account for testing" set the same email address you use for logging in to HC after you add the integration in HA.

**Comment by iceymanz** [comment]
FWIW, the developer portal does not appear to have moved to singlekey. Only the consumer phone application. These have always been two seperate logins. Once you have the API key, that's the only form of auth ekutners app is using. 

For some reason the developer portal does seem to forget my password and every time I log back in I need to reset it but that's an unrelated issue.

**Comment by socialhugo** [comment]
I finally got it working again! I deleted my account as home connect user in the android app. I deleted my account as developer via de webportal. And then I registered/created both accounts again. Using the same e-mail address ofcourse.

I also got a reply from the home connect helpdesk. I explained my situation, but they say 'nothing' changed. But something has changed otherwise it would still work now wouldn't it :)

Thanks for thinking along
